crates/wascc-provider/src/states/finished.rs

```diff
@@ -0,0 +1,20 @@
+use kubelet::state::{State, Transition};
+use kubelet::{
+    pod::{Phase, Pod},
+    state,
+};
+
+use crate::{make_status, PodState};
+
+state!(
+    /// Pod execution completed with no errors.
+    Finished,
+    PodState,
+    Finished,
+    Finished,
+    {
+        // TODO: Wait for deleted.
```

Comment:
Should "Finished" just be a pseudo-state always included by the krustlet crate itself?

Reply:
Hmm, maybe so it would transition to that to exit? Would we also provide failed or terminated or something like that?

Reply:
Either one would work I think. My personal preference would be to have a "return `Completed` to exit". Either way helps avoid provider implementations having to do the final k8s waiting/cleanup

Reply:
Ah, I see. so this is actually handled in the Pod::queue, I need to remove this comment.